### PR TITLE
Add ScalaDoc to Quasiquotes and Liftables parts of api

### PR DESCRIPTION
**Pull request by densh**
review @xeno-by @heathermiller

### CODE REVIEW

File: src/reflect/scala/reflect/api/Liftables.scala [hunk 1]
```diff
@@ -2,27 +2,72 @@ package scala
...
+     *
+     *  {{{
+     *  scala> object O
+     *
+     *  scala> val Oref = symbolOf[O.type].asClass.module
+     *
+     *  scala> implicit val liftO = Liftable[O.type] { _ => q"$Oref" }
+     *
+     *  scala> q"$O"
+     *  res16: universe.Tree = O
```
Comment (xeno-by) [inline src/reflect/scala/reflect/api/Liftables.scala]: res16 :)

File: src/reflect/scala/reflect/api/Quasiquotes.scala [hunk 2]
```diff
@@ -3,10 +3,16 @@ package api
 
 trait Quasiquotes { self: Universe =>
 
-  // implementation is hardwired to `dispatch` method of `scala.tools.reflect.quasiquotes.Quasiquotes`
-  // using the mechanism implemented in `scala.tools.reflect.FastTrack`
+  /** Implicit class that introduced `q`, `tq`, `cq,` `p` and `fq` string interpolators
```
Comment (xeno-by) [inline src/reflect/scala/reflect/api/Quasiquotes.scala]: introduces?

File: src/reflect/scala/reflect/api/Liftables.scala (hunk 1 above)
Comment (densh) [inline src/reflect/scala/reflect/api/Liftables.scala]: Fixed.

File: src/reflect/scala/reflect/api/Quasiquotes.scala (hunk 2 above)
Comment (densh) [inline src/reflect/scala/reflect/api/Quasiquotes.scala]: Fixed.

**Comment by xeno-by** [comment]
LGTM

**Pull request merged**